crates/primitives/src/transaction/pooled.rs

```diff
@@ -581,23 +360,110 @@ impl Decodable for PooledTransactionsElement {
         // Check if the tx is a list
         if header.list {
```

Comment:
Ideally we would just delegate to `Decodable2718::network_decode` here as it's done for `TransactionSigned`. However, this `decode` is more strict as it can't decode enveloped transactions (without header):
https://github.com/paradigmxyz/reth/blob/227e2933907981f5effc5ed162b61618dba6a28a/crates/primitives/src/transaction/mod.rs#L1335-L1337

I am wondering if this should be supported by plain rlp decoding at all? feels a bit unsafe and I believe there is always a correct encoding format defined for each context? I guess this is only useful for user-facing methods eg `sendRawTransaction` where we'd want to give more flexibility and in those cases fallback to enveloped decoding can be done explicitly in-place or through a helper fn

Reply:
yep, I think `decode_2718` is already strict, just need to remove this workaround from `network_decode` https://github.com/alloy-rs/alloy/blob/47e57b3b47d0dc300a2dbbf75ebe7dbd8b644939/crates/eips/src/eip2718.rs#L154-L159